### PR TITLE
No stream mode in v2 chat

### PR DESCRIPTION
**Pull request by neoxelox**
Closes: https://github.com/latitude-dev/latitude-llm/issues/535

Implements:
- `stream: false` option in `/chat` v2 endpoint that returns the last chain event as a JSON response.
- `chat` sdk function now accepts `stream` as a parameter

Notice that as `stream` option is now `false` by default, this is a breaking change for both API and SDK. But as discussed by the team this is ok,

### CODE REVIEW

File: docs/guides/prompt-manager/api-access.mdx [hunk 1]
```diff
@@ -158,12 +158,15 @@ The previously described `POST /projects/{projectId}/versions/{versionUuid}/docu
...
+  ],
+  "stream": true
 }
 ```
 
+- `stream`: Optional boolean parameter (defaults to `false`). When set to true, the response will be a stream of Server-Sent Events (SSE). If false, a single JSON response containing the last event is returned.
+
 **Response:**
-The response is a stream of Server-Sent Events (SSE), similar to the "Run a Document" endpoint.
+The response is a stream of Server-Sent Events (SSE) or a single JSON response containing the final event, similar to the "Run a Document" endpoint.
```
Comment (csansoon) [inline docs/guides/prompt-manager/api-access.mdx:170]: nicee

**Review by csansoon — DISMISSED** [review]

File: apps/gateway/src/routes/api/v2/conversations/[conversationUuid]/handlers/chat.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            captureException(error)
+          }
+
+          return Promise.resolve()
+        },
+      )
+    }
+
+    const response = (await result.response).unwrap()
+    const body = documentRunPresenter(response).unwrap()
```
Comment (andresgutgon) [inline apps/gateway/src/routes/api/v2/conversations/[conversationUuid]/handlers/chat.ts:70]: Is chat response the same as a run response?

File: apps/gateway/src/routes/api/v2/conversations/[conversationUuid]/handlers/chat.test.ts [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+        errorCode: 'LatitudeError',
+        message: 'Conversation messages not found in response',
+        details: {},
+      })
+      expect(mocks.captureException).toHaveBeenCalledWith(
+        new LatitudeError('Conversation messages not found in response'),
+      )
+    })
+  })
+})
```
Comment (geclos) [inline apps/gateway/src/routes/api/v2/conversations/[conversationUuid]/handlers/chat.test.ts:546]: beautiful

File: apps/gateway/src/routes/api/v2/conversations/[conversationUuid]/handlers/chat.ts (hunk 2 above)
Comment (geclos) [reply apps/gateway/src/routes/api/v2/conversations/[conversationUuid]/handlers/chat.ts:70]: it should be yes

File: packages/core/src/services/providerLogs/addMessages.ts [hunk 4]
```diff
@@ -62,13 +68,11 @@ export async function addMessages({
     )
   }
 
-  let responseResolve: (value?: ChainStepResponse<StreamType>) => void
+  let responseResolve: (value: ChainResponse<StreamType>) => void
```
Comment (geclos) [inline packages/core/src/services/providerLogs/addMessages.ts:71]: hmm you changed the response type of this service yet no test is failing, maybe a good opportunity to add tests for this service.

**Review by geclos — APPROVED** [review]
good jod!

**Review by andresgutgon — DISMISSED** [review]

File: packages/sdks/typescript/src/tests/helpers/chat.ts [hunk 5]
```diff
@@ -8,27 +9,43 @@ import { vi } from 'vitest'
...
+  const mockAuthHeader = vi.fn()
+  const mockUrl = vi.fn()
+  const mockBody = vi.fn()
+  let body = {}
   server.use(
     http.post(
-      `http://localhost:8787/api/${apiVersion}/conversations/fake-document-log-uuid/chat`,
-      (info) => {
-        mockFn(info.request.headers.get('Authorization'))
+      `http://localhost:8787/api/${apiVersion}/conversations/${docPath}/chat`,
```
Comment (geclos) [inline packages/sdks/typescript/src/tests/helpers/chat.ts:27]: docPath? this is the conversation uuid no?

**Review by geclos — COMMENTED** [review]
update the sdk's beta version in package.json and 👌🏼

File: packages/sdks/typescript/src/tests/helpers/chat.ts (hunk 5 above)
Comment (neoxelox) [reply packages/sdks/typescript/src/tests/helpers/chat.ts:27]: true

**Review by csansoon — APPROVED** [review]
Looks good! 🎉

**Pull request merged**